--- ./doc/book/Makefile.orig	2004-07-07 18:55:28.000000000 -0500
+++ ./doc/book/Makefile	2004-07-07 18:58:59.000000000 -0500
@@ -68,16 +68,7 @@
 $(BOOK_VERSION_SOURCE): book-version
 
 book-version:
-	@if $(SVNVERSION) . > /dev/null; then \
-	echo '<!ENTITY svn.version "Revision '`$(SVNVERSION) .`'">' > $(BOOK_VERSION_SOURCE).tmp; \
-	else \
-	echo '<!ENTITY svn.version "">' > $(BOOK_VERSION_SOURCE).tmp; \
-	fi
-	@if cmp -s $(BOOK_VERSION_SOURCE) $(BOOK_VERSION_SOURCE).tmp; then \
-	rm $(BOOK_VERSION_SOURCE).tmp; \
-	else \
-	mv $(BOOK_VERSION_SOURCE).tmp $(BOOK_VERSION_SOURCE); \
-	fi
+	echo '<!ENTITY svn.version "Version '`cat book/package.version`'">' > $(BOOK_VERSION_SOURCE);
 
 book-html: $(BOOK_HTML_TARGET)
 
