--- subversion-0.23.0/build-outputs.mk 2003-06-14 11:00:54     1.1
+++ subversion-0.23.0/build-outputs.mk 2003-06-14 11:22:19
@@ -339,6 +339,6 @@
 
 install-mods-shared: subversion/mod_authz_svn/mod_authz_svn.la subversion/mod_dav_svn/mod_dav_svn.la
-	cd subversion/mod_authz_svn ; $(MKDIR) "$(APACHE_LIBEXECDIR)" ; $(INSTALL_MOD_SHARED) -n authz_svn mod_authz_svn.la
-	cd subversion/mod_dav_svn ; $(MKDIR) "$(APACHE_LIBEXECDIR)" ; $(INSTALL_MOD_SHARED) -n dav_svn mod_dav_svn.la
+	cd subversion/mod_authz_svn ; mkdir -p $(RPM_BUILD_ROOT)/etc/httpd/2.0/modules; cp ./.libs/mod_authz_svn.so ./mod_authz_svn.la $(RPM_BUILD_ROOT)/etc/httpd/2.0/modules
+	cd subversion/mod_dav_svn ; mkdir -p $(RPM_BUILD_ROOT)/etc/httpd/2.0/modules; cp ./.libs/mod_dav_svn.so ./mod_dav_svn.la $(RPM_BUILD_ROOT)/etc/httpd/2.0/modules
 
 install-mods-static: subversion/libsvn_fs/libsvn_fs-1.la-a subversion/mod_dav_svn/mod_dav_svn.la-a subversion/libsvn_delta/libsvn_delta-1.la-a subversion/mod_authz_svn/mod_authz_svn.la-a subversion/libsvn_repos/libsvn_repos-1.la-a subversion/libsvn_subr/libsvn_subr-1.la-a subversion/mod_dav_svn/static/config.m4 subversion/mod_dav_svn/static/Makefile.in
