diff -rbuN subversion-0.29.0/build/generator/gen_make.py subversion-0.29.0-patch/build/generator/gen_make.py
--- subversion-0.29.0/build/generator/gen_make.py	2003-09-17 19:36:40.000000000 -0400
+++ subversion-0.29.0-patch/build/generator/gen_make.py	2003-09-17 21:11:58.000000000 -0400
@@ -71,13 +71,16 @@
         '%s_DEPS = %s %s\n'
         '%s_OBJECTS = %s\n'
         '%s: $(%s_DEPS)\n'
-        '\tcd %s && %s -o %s $(%s_OBJECTS) %s $(LIBS)\n\n'
+        '\t@echo LINK %s\n'
+        '\t@(cd %s && %s -o %s $(%s_OBJECTS) %s $(LIBS))\n\n'
         % (targ_varname, string.join(objects + deps), target_ob.add_deps,
 
            targ_varname, objnames,
 
            target_ob.filename, targ_varname,
 
+	   target_ob.filename,
+
            path, target_ob.link_cmd, os.path.basename(target_ob.filename),
            targ_varname, string.join(libs))
         )
@@ -114,10 +117,11 @@
           dirname, fname = os.path.split(file)
           base, ext = os.path.splitext(fname)
           name = string.replace(base, 'mod_', '')
-          self.ofile.write('\tcd %s ; '
-                           '$(MKDIR) "$(APACHE_LIBEXECDIR)" ; '
-                           '$(INSTALL_MOD_SHARED) -n %s %s\n'
-                           % (dirname, name, fname))
+          self.ofile.write('\t@echo LINK %s\n'
+                           '\t@(cd %s ; '
+                           '$(MKDIR) "$(APACHE_LIBEXECDIR)" ; '
+                           '$(INSTALL_MOD_SHARED) -n %s %s)\n'
+                           % (fname, dirname, name, fname))
           if ext == '.la':
             la_tweaked[file + '-a'] = None
 
@@ -132,15 +136,19 @@
         # Construct a .libs directory within the Apache area and populate it
         # with the appropriate files. Also drop the .la file in the target dir.
         self.ofile.write('\ninstall-mods-static: %s\n'
-                         '\t$(MKDIR) $(DESTDIR)%s\n'
+                         '\t@echo MKDIR %s\n'
+                         '\t@($(MKDIR) $(DESTDIR)%s)\n'
                          % (string.join(la_tweaked + self.apache_files),
+                            os.path.join('$(APACHE_TARGET)', '.libs'),
                             os.path.join('$(APACHE_TARGET)', '.libs')))
         for file in la_tweaked:
           dirname, fname = os.path.split(file)
           base = os.path.splitext(fname)[0]
-          self.ofile.write('\t$(INSTALL_MOD_STATIC) %s $(DESTDIR)%s\n'
-                           '\t$(INSTALL_MOD_STATIC) %s $(DESTDIR)%s\n'
-                           % (os.path.join(dirname, '.libs', base + '.a'),
+          self.ofile.write('\t@echo LINSTALL %s\n'
+                           '\t@($(INSTALL_MOD_STATIC) %s $(DESTDIR)%s)\n'
+                           '\t@($(INSTALL_MOD_STATIC) %s $(DESTDIR)%s)\n'
+                           % (base,
+                              os.path.join(dirname, '.libs', base + '.a'),
                               os.path.join('$(APACHE_TARGET)',
                                            '.libs',
                                            base + '.a'),
@@ -149,8 +157,10 @@

         # copy the other files to the target dir
         for file in self.apache_files:
-          self.ofile.write('\t$(INSTALL_MOD_STATIC) %s $(DESTDIR)%s\n'
-                           % (file, os.path.join('$(APACHE_TARGET)',
+          self.ofile.write('\t@echo INSTALL %s\n'
+                           '\t@($(INSTALL_MOD_STATIC) %s $(DESTDIR)%s)\n'
+                           % (file, 
+                              file, os.path.join('$(APACHE_TARGET)',
                                                  os.path.basename(file))))
         self.ofile.write('\n')
 
@@ -158,13 +168,16 @@
         area_var = string.replace(area, '-', '_')
         upper_var = string.upper(area_var)
         self.ofile.write('install-%s: %s\n'
-                         '\t$(MKDIR) $(DESTDIR)$(%sdir)\n'
-                         % (area, string.join(files), area_var))
+                         '\t@echo MKDIR %s\n'
+                         '\t@($(MKDIR) $(DESTDIR)$(%sdir))\n'
+                         % (area, string.join(files),area, area_var))
         for file in files:
           # cd to dirname before install to work around libtool 1.4.2 bug.
           dirname, fname = os.path.split(file)
-          self.ofile.write('\tcd %s ; $(INSTALL_%s) %s $(DESTDIR)%s\n'
-                           % (dirname,
+          self.ofile.write('\t@echo INSTALL %s\n'
+                           '\t@(cd %s ; $(INSTALL_%s) %s $(DESTDIR)%s)\n'
+                           % (fname,
+                              dirname,
                               upper_var,
                               fname,
                               os.path.join('$(%sdir)' % area_var, fname)))
@@ -176,11 +189,13 @@
     includedir = os.path.join('$(includedir)',
                               'subversion-%s' % self.cfg.version)
     self.ofile.write('install-include: %s\n'
-                     '\t$(MKDIR) $(DESTDIR)%s\n'
-                     % (string.join(self.includes), includedir))
+                     '\t@echo MKDIR %s\n'
+                     '\t@($(MKDIR) $(DESTDIR)%s)\n'
+                     % (includedir, string.join(self.includes), includedir))
     for file in self.includes:
-      self.ofile.write('\t$(INSTALL_INCLUDE) %s $(DESTDIR)%s\n'
-                       % (os.path.join('$(top_srcdir)', file),
+      self.ofile.write('\t@echo INSTALL %s\n'
+                       '\t@($(INSTALL_INCLUDE) %s $(DESTDIR)%s)\n'
+                       % (file, os.path.join('$(top_srcdir)', file),
                           os.path.join(includedir, os.path.basename(file))))
 
     self.ofile.write('\n# handy shortcut targets\n')
@@ -204,8 +219,10 @@
 
     for objname, sources in self.graph.get_deps(gen_base.DT_SWIG_C):
       deps = string.join(map(str, sources))
-      self.ofile.write('%s: %s\n\t$(RUN_SWIG_%s) %s\n'
-                       % (objname, deps, string.upper(objname.lang_abbrev),
+      self.ofile.write('%s: %s\n'
+                       '\t@echo SWIG %s\n'
+                       '\t@($(RUN_SWIG_%s) %s)\n'
+                       % (objname, deps, objname, string.upper(objname.lang_abbrev),
                           os.path.join('$(top_srcdir)', str(sources[0]))))
 
     for objname, sources in self.graph.get_deps(gen_base.DT_OBJECT):
@@ -214,10 +231,10 @@
       cmd = getattr(objname, 'build_cmd', '')
       if cmd:
         if not getattr(objname, 'source_generated', 0):
-          self.ofile.write('\t%s %s\n' % (cmd, os.path.join('$(top_srcdir)',
+          self.ofile.write('\t@echo APACHECC %s\n\t@(%s %s)\n' % (sources[0],cmd, os.path.join('$(top_srcdir)',
                                                             str(sources[0]))))
         else:
-          self.ofile.write('\t%s %s\n' % (cmd, sources[0]))
+          self.ofile.write('\t@echo SWIGCC %s\n\t@(%s %s)\n' % (sources[0],cmd, sources[0]))
 
   def write_symbols(self):
     wrappers = { }
diff -rbuN subversion-0.29.0/Makefile.in subversion-0.29.0-patch/Makefile.in
--- subversion-0.29.0/Makefile.in	2003-09-17 19:33:51.000000000 -0400
+++ subversion-0.29.0-patch/Makefile.in	2003-09-17 21:22:24.000000000 -0400
@@ -159,11 +159,11 @@
 
 
 INSTALL = @INSTALL@
-INSTALL_LIB = $(LIBTOOL) --mode=install $(INSTALL)
+INSTALL_LIB = $(LIBTOOL) $(LTFLAGS) --mode=install $(INSTALL)
 INSTALL_BASE_LIB = $(INSTALL_LIB)
 INSTALL_FS_LIB = $(INSTALL_LIB)
 INSTALL_DAV_LIB = $(INSTALL_LIB)
-INSTALL_BIN = $(LIBTOOL) --mode=install $(INSTALL)
+INSTALL_BIN = $(LIBTOOL) $(LTFLAGS) --mode=install $(INSTALL)
 INSTALL_FS_BIN = $(INSTALL_BIN)
 INSTALL_INCLUDE = $(INSTALL) -m 644
 INSTALL_MOD_SHARED = @APXS@ -i -S LIBEXECDIR="$(APACHE_LIBEXECDIR)" @MOD_ACTIVATION@
@@ -212,20 +212,23 @@
 # clean everything but the bulky test output, returning the system back
 # to before 'make' was run.
 fast-clean: doc-clean
-	@list='$(BUILD_DIRS)'; for i in $$list; do                   \
+	@echo Quick clean
+	@(list='$(BUILD_DIRS)'; for i in $$list; do                   \
             echo "Cleaning $$i ..." ;                                \
             (cd $$i && rm -f *.o *.lo *.la *.la-a && rm -rf .libs) ; \
-        done
-	rm -f $(CLEAN_FILES)
+        done)
+	@rm -f $(CLEAN_FILES)
 
 # clean all but bulky test output, returning to before './configure' was run.
 fast-distclean: fast-clean
-	rm -f Makefile config.cache config.log config.status libtool \
+	@echo Quick distribution clean
+	@rm -f Makefile config.cache config.log config.status libtool \
               svn_private_config.h svn-config
 
 # clean everything out, returning to before './autogen.sh' was run.
 local-extraclean: local-distclean
-	rm -f build-outputs.mk svn_private_config.h.in configure \
+	@echo Restoring pristine distribution
+	@rm -f build-outputs.mk svn_private_config.h.in configure \
               ac-helpers/config.guess ac-helpers/config.sub      \
               ac-helpers/libtool.m4 ac-helpers/ltconfig          \
               ac-helpers/ltmain.sh
@@ -240,7 +243,8 @@
 
 ### HACK!! Find a better way to do this
 revision-install:
-	subversion/svnversion/svnversion $(top_srcdir) /repos/svn/trunk > $(DESTDIR)$(includedir)/subversion-1/svn-revision.txt
+	@echo Recording current revision
+	@subversion/svnversion/svnversion $(top_srcdir) /repos/svn/trunk > $(DESTDIR)$(includedir)/subversion-1/svn-revision.txt
 
 install-static: @INSTALL_STATIC_RULES@
 
@@ -248,7 +252,8 @@
 # "make check TESTS=subversion/tests/clients/cmdline/basic_tests.py"
 #  will perform only basic tests (likewise for other tests).
 check: $(TEST_DEPS) @FS_TEST_DEPS@
-	@$(PYTHON) $(top_srcdir)/build/pycheck.py;                         \
+	@echo Cleaning tests
+	@($(PYTHON) $(top_srcdir)/build/pycheck.py;                         \
 	flags="--verbose";                                                 \
 	if test "$(CLEANUP)" != ""; then                                   \
 	  flags="--cleanup $$flags";                                       \
@@ -258,7 +263,7 @@
 	fi;                                                                \
 	$(PYTHON) $(top_srcdir)/build/run_tests.py $$flags                 \
 		  '$(abs_srcdir)' '$(abs_builddir)' '$(PYTHON)' '$(SHELL)' \
-		  $(TESTS)
+		  $(TESTS))
 
 # First, set up Apache as documented in
 # subversion/tests/clients/cmdline/README.
@@ -277,7 +282,8 @@
 	  BASE_URL=svn+ssh://localhost`pwd`/subversion/tests/clients/cmdline
 
 check-clean:
-	rm -rf subversion/tests/clients/cmdline/repositories   \
+	@echo Cleaning checks
+	@rm -rf subversion/tests/clients/cmdline/repositories   \
                subversion/tests/clients/cmdline/working_copies \
                subversion/tests/clients/cmdline/local_tmp      \
                subversion/tests/libsvn_fs/test-repo-*          \
@@ -286,26 +292,26 @@
                tests.log
 
 mkdir-init:
-	@list='$(BUILD_DIRS) $(DOC_DIRS) $(SWIG_DIRS)'; \
+	@(list='$(BUILD_DIRS) $(DOC_DIRS) $(SWIG_DIRS)'; \
         for i in $$list doc; do                         \
             echo "mkdir $$i" ;                          \
             $(MKDIR) $$i ;                              \
-        done
+        done)
 
 # recursive targets to handle external projects (e.g. APR, Neon)
 external-all external-install:
-	@target=`echo $@ | sed s/external-//`;          \
+	@(target=`echo $@ | sed s/external-//`;          \
         list='$(EXTERNAL_PROJECT_DIRS)';                \
         for i in $$list; do                             \
             echo "------ making $$target in $$i";       \
             (cd $$i && $(MAKE) $$target) || exit 1;     \
             echo "------ completed $$target in $$i";    \
-        done;
+        done;)
 
 ### apr-util relies on generated files in apr, so reverse the
 ### list so that it is cleaned before apr.
 external-clean external-distclean:
-	@target=`echo $@ | sed s/external-//`;          \
+	@(target=`echo $@ | sed s/external-//`;          \
         list='$(EXTERNAL_PROJECT_DIRS)';                \
         for i in $$list; do                             \
             revlist="$$i $$revlist";                    \
@@ -314,12 +320,12 @@
             echo "------ making $$target in $$i";       \
             (cd $$i && $(MAKE) $$target) || exit 1;     \
             echo "------ completed $$target in $$i";    \
-        done;
+        done;)
 
 ### temporary hack. Neon does not have an "extraclean" and neither does db
 ### If we don't have extraclean -- do the next best thing.
 external-extraclean:
-	@list='$(EXTERNAL_PROJECT_DIRS)';                                \
+	@(list='$(EXTERNAL_PROJECT_DIRS)';                                \
         for i in $$list; do                                              \
             revlist="$$i $$revlist";                                     \
         done;                                                            \
@@ -333,7 +339,7 @@
                 (cd $$i && $(MAKE) extraclean) || exit 1;                \
                 echo "------ completed extraclean in $$i";               \
             fi;                                                          \
-        done;
+        done;)
 
 
 # DOCUMENTATION RULES
@@ -364,31 +370,38 @@
 # Different outputs for the main Subversion 'book', written in docbook.
 # Assumes you have {xsltproc, fop, stylesheets} installed.
 doc-book-html:
-	cd ${BOOK_TOP} && $(MAKE) book-html
+	@echo Make html book
+	@cd ${BOOK_TOP} && $(MAKE) book-html
 doc-book-pdf:
-	cd ${BOOK_TOP} && $(MAKE) book-pdf
+	@echo Make pdf book
+	@cd ${BOOK_TOP} && $(MAKE) book-pdf
 doc-book-ps:
-	cd ${BOOK_TOP} && $(MAKE) book-ps
+	@echo Make postscript book
+	@cd ${BOOK_TOP} && $(MAKE) book-ps
 
 # Different outputs for the misc docs.
 # Assumes you have {xsltproc, fop, stylesheets} installed.
 doc-misc-docs-html:
-	cd ${BOOK_TOP} && $(MAKE) misc-docs-html
+	@echo Make html docs
+	@cd ${BOOK_TOP} && $(MAKE) misc-docs-html
 doc-misc-docs-pdf:
-	cd ${BOOK_TOP} && $(MAKE) misc-docs-pdf
+	@echo Make pdf docs
+	@cd ${BOOK_TOP} && $(MAKE) misc-docs-pdf
 doc-misc-docs-ps:
-	cd ${BOOK_TOP} && $(MAKE) misc-docs-ps
+	@echo Make postscripts docs
+	@cd ${BOOK_TOP} && $(MAKE) misc-docs-ps
 
 
 doc-clean:
-	for d in $(DOC_DIRS);                               \
+	@echo Clean docs
+	@(for d in $(DOC_DIRS);                               \
         do                                                  \
             (cd $$d;                                        \
              rm -f *.info *.info-[1-9]                      \
                    *.aux *.cp *.fn *.ky *.log *.pg *.toc    \
                    *.tp *.vr                                \
                    *.dvi *.txt *.html *.ps *.pdf *.fo);     \
-        done
+        done)
 
 java-api:
 	@$(MAKE) -C $(abs_srcdir)/subversion/bindings/java "JAVAC=$(JAVAC)"
@@ -401,38 +414,48 @@
 .SUFFIXES: .c .lo .o .la-a .la .texi .info .dvi .txt .html .ps .pdf
 
 .c.o:
-	$(COMPILE) -o $@ -c $<
+	@echo CC $<
+	@$(COMPILE) -o $@ -c $<
 
 .c.lo:
-	$(LT_COMPILE) -o $@ -c $<
+	@echo LCC $<
+	@$(LT_COMPILE) -o $@ -c $<
 
 .la.la-a:
-	sed "/library_names/s/'.*'/''/" $< > $@
+	@echo SED $<
+	@sed "/library_names/s/'.*'/''/" $< > $@
 
 
 .texi.info:
-	$(MAKEINFO) -I `dirname $<` -o $@ $< 
+	@echo INFO $<
+	@$(MAKEINFO) -I `dirname $<` -o $@ $< 
 
 .texi.dvi:
-	wd=`pwd` ; cd `dirname $@` ; $(TEXI2DVI) $$wd/$<
+	@echo DVI $<
+	@wd=`pwd` ; cd `dirname $@` ; $(TEXI2DVI) $$wd/$<
 
 .texi.txt:
-	$(MAKEINFO) -I `dirname $<` --no-headers -o $@ $<
+	@echo TXT $<
+	@$(MAKEINFO) -I `dirname $<` --no-headers -o $@ $<
 
 .texi.html:
-	$(MAKEINFO) -I `dirname $<` --no-split --html -o $@ $<
+	@echo HTML $<
+	@$(MAKEINFO) -I `dirname $<` --no-split --html -o $@ $<
 
 .dvi.ps:
-	$(DVIPS) $< -o $@
+	@echo PS $<
+	@$(DVIPS) $< -o $@
 
 .dvi.pdf:
-	$(DVIPDF) $< $@
+	@echo PDF $<
+	@$(DVIPDF) $< $@
 
 install-docs: install-man @INSTALL_INFO@
 
 manroot = $(mandir)/man
 install-man:
-	@list='$(MANPAGES)';                                               \
+	@echo Install manpages
+	@(list='$(MANPAGES)';                                               \
         for i in $$list; do                                                \
           if test -f $(srcdir)/$$i; then file=$(srcdir)/$$i;               \
           else file=$$i; fi;                                               \
@@ -443,15 +466,16 @@
           inst=`echo $$inst`.$$ext;                                        \
           echo "$(INSTALL_DATA) $$file $(DESTDIR)$(manroot)$$ext/$$inst";  \
           $(INSTALL_DATA) $$file $(DESTDIR)$(manroot)$$ext/$$inst;         \
-        done
+        done)
 
 install-info: doc-all-info
-	$(MKDIR) $(DESTDIR)$(infodir)
-	@list='$(INFOPAGES)';                                         \
+	@ echo Install docs
+	@$(MKDIR) $(DESTDIR)$(infodir)
+	@(list='$(INFOPAGES)';                                         \
         for i in $$list; do                                           \
           if test -f $(srcdir)/$$i; then file=$(srcdir)/$$i;          \
           else file=$$i; fi;                                          \
           inst=`echo $$file | sed -e 's/^.*\///'`;                    \
           echo "$(INSTALL_DATA) $$file $(DESTDIR)$(infodir)/$$inst";  \
           $(INSTALL_DATA) $$file $(DESTDIR)$(infodir)/$$inst;         \
-        done
+        done)
