--- subversion-M5/build-outputs.mk	2001/10/28 20:08:47	1.1
+++ subversion-M5/build-outputs.mk	2001/10/28 20:10:19
@@ -208,7 +208,8 @@
 	cd subversion/svnadmin ; $(INSTALL_FS_BIN) svnadmin $(fs_bindir)/svnadmin
 
 install-mods-shared: subversion/mod_dav_svn/libmod_dav_svn.la
-	cd subversion/mod_dav_svn ; $(INSTALL_MOD_SHARED) -n dav_svn libmod_dav_svn.la
+	cd subversion/mod_dav_svn ; ($(INSTALL_MOD_SHARED) -n dav_svn libmod_dav_svn.la || true); mkdir -p $(libexecdir); cp ./.libs/libmod_dav_svn.so $(libexecdir)/libmod_dav_svn.so; cp ./libmod_dav_svn.la $(libexecdir)
+#$(INSTALL_MOD_SHARED) -n dav_svn libmod_dav_svn.la
 
 install-mods-static: subversion/libsvn_delta/libsvn_delta.la-a subversion/libsvn_fs/libsvn_fs.la-a subversion/libsvn_subr/libsvn_subr.la-a subversion/mod_dav_svn/libmod_dav_svn.la-a subversion/libsvn_repos/libsvn_repos.la-a subversion/mod_dav_svn/static/config.m4 subversion/mod_dav_svn/static/Makefile.in
 	$(MKDIR) $(APACHE_TARGET)/.libs
