--- subversion-0.14.3/subversion/bindings/swig/python/setup.py	2002/09/21 17:45:50	1.1
+++ subversion-0.14.3/subversion/bindings/swig/python/setup.py	2002/09/21 17:47:13
@@ -21,8 +21,8 @@
 from distutils.command import build_ext
 from distutils import dir_util
 
-INC_DIRS=['..', '../../../include', '../../../../apr/include']
-LIB_DIRS=['/usr/local/svn/lib']
+INC_DIRS=['..', '../../../include', '/usr/local/apache2/include']
+LIB_DIRS=['/usr/local/apache2/lib']
 
 class build_swig(build_ext.build_ext):
   def initialize_options(self):
@@ -67,47 +67,44 @@
            ext_modules=[
              core.Extension("_client",
                             ["../svn_client.i"],
-                            libraries=['svn_client-1', 'svn_swig_py-1',
-                                       'swigpy'],
+                            libraries=['svn_client-1', 'svn_swig_py-1'],
                             library_dirs=LIB_DIRS,
                             ),
              core.Extension("_delta",
                             ["../svn_delta.i"],
-                            libraries=['svn_delta-1', 'svn_swig_py-1',
-                                       'swigpy'],
+                            libraries=['svn_delta-1', 'svn_swig_py-1'],
                             library_dirs=LIB_DIRS,
                             ),
              core.Extension("_fs",
                             ["../svn_fs.i"],
-                            libraries=['svn_fs-1', 'svn_swig_py-1', 'swigpy'],
+                            libraries=['svn_fs-1', 'svn_swig_py-1'],
                             library_dirs=LIB_DIRS,
                             ),
              core.Extension("_ra",
                             ["../svn_ra.i"],
-                            libraries=['svn_ra-1', 'swigpy'],
+                            libraries=['svn_ra-1'],
                             library_dirs=LIB_DIRS,
                             ),
              core.Extension("_repos",
                             ["../svn_repos.i"],
-                            libraries=['svn_repos-1', 'svn_swig_py-1',
-                                       'swigpy'],
+                            libraries=['svn_repos-1', 'svn_swig_py-1'],
                             library_dirs=LIB_DIRS,
                             ),
              core.Extension("_wc",
                             ["../svn_wc.i"],
-                            libraries=['svn_wc-1', 'svn_swig_py-1', 'swigpy'],
+                            libraries=['svn_wc-1', 'svn_swig_py-1'],
                             library_dirs=LIB_DIRS,
                             ),
              core.Extension("_util",
                             ["../util.i"],
-                            libraries=['svn_subr-1', 'swigpy', 'apr-0'],
+                            libraries=['svn_subr-1', 'apr-0'],
                             library_dirs=LIB_DIRS,
                             ),
 
              ### will 'auth' be its own, or bundled elsewhere?
              #core.Extension("_auth",
              #               ["../svn_auth.i"],
-             #               libraries=['svn_subr-1', 'swigpy', 'apr-0'],
+             #               libraries=['svn_subr-1', 'apr-0'],
              #               library_dirs=LIB_DIRS,
              #               ),
              ],
