--- subversion-0.23.0/build-outputs.mk	2003/06/11 02:58:05	1.1
+++ subversion-0.23.0/build-outputs.mk	2003/06/11 03:00:37
@@ -339,7 +339,7 @@
 
 install-mods-shared: subversion/mod_authz_svn/mod_authz_svn.la subversion/mod_dav_svn/mod_dav_svn.la
-	cd subversion/mod_authz_svn ; $(MKDIR) "$(APACHE_LIBEXECDIR)" ; $(INSTALL_MOD_SHARED) -n authz_svn mod_authz_svn.la
-	cd subversion/mod_dav_svn ; $(MKDIR) "$(APACHE_LIBEXECDIR)" ; $(INSTALL_MOD_SHARED) -n dav_svn mod_dav_svn.la
+	cd subversion/mod_authz_svn ; $(MKDIR) $(RPM_BUILD_ROOT)/usr/local/apache2/modules; cp ./.libs/mod_authz_svn.so ./mod_authz_svn.la $(RPM_BUILD_ROOT)/usr/local/apache2/modules
+	cd subversion/mod_dav_svn ; $(MKDIR) $(RPM_BUILD_ROOT)/usr/local/apache2/modules; cp ./.libs/mod_dav_svn.so ./mod_dav_svn.la $(RPM_BUILD_ROOT)/usr/local/apache2/modules
 
 install-mods-static: subversion/libsvn_fs/libsvn_fs-1.la-a subversion/mod_dav_svn/mod_dav_svn.la-a subversion/libsvn_delta/libsvn_delta-1.la-a subversion/mod_authz_svn/mod_authz_svn.la-a subversion/libsvn_repos/libsvn_repos-1.la-a subversion/libsvn_subr/libsvn_subr-1.la-a subversion/mod_dav_svn/static/config.m4 subversion/mod_dav_svn/static/Makefile.in
 	$(MKDIR) $(DESTDIR)$(APACHE_TARGET)/.libs
