--- subversion-0.14.2/build-outputs.mk	2002/09/07 15:52:24	1.1
+++ subversion-0.14.2/build-outputs.mk	2002/09/07 15:56:16
@@ -248,7 +248,8 @@
 	cd subversion/svnlook ; $(INSTALL_FS_BIN) svnlook $(DESTDIR)$(fs_bindir)/svnlook
 
 install-mods-shared: subversion/mod_dav_svn/mod_dav_svn.la
-	cd subversion/mod_dav_svn ; apxs_libexecdir="$(DESTDIR)`$(APXS) -q LIBEXECDIR`" ; $(MKDIR) "$$apxs_libexecdir" ; $(INSTALL_MOD_SHARED) -S LIBEXECDIR="$$apxs_libexecdir" -n dav_svn mod_dav_svn.la
+	cd subversion/mod_dav_svn ; mkdir -p $(RPM_BUILD_ROOT)/usr/local/apache2/modules; cp ./.libs/mod_dav_svn.so ./mod_dav_svn.la $(RPM_BUILD_ROOT)/usr/local/apache2/modules/
+# apxs_libexecdir="$(DESTDIR)`$(APXS) -q LIBEXECDIR`" ; $(MKDIR) "$$apxs_libexecdir" ; $(INSTALL_MOD_SHARED) -S LIBEXECDIR="$$apxs_libexecdir" -n dav_svn mod_dav_svn.la
 
 install-mods-static: subversion/libsvn_subr/libsvn_subr-1.la-a subversion/libsvn_delta/libsvn_delta-1.la-a subversion/mod_dav_svn/mod_dav_svn.la-a subversion/libsvn_repos/libsvn_repos-1.la-a subversion/libsvn_fs/libsvn_fs-1.la-a subversion/mod_dav_svn/static/config.m4 subversion/mod_dav_svn/static/Makefile.in
 	$(MKDIR) $(DESTDIR)$(APACHE_TARGET)/.libs
