diff -rbuN subversion-0.29.0/build/ac-macros/svn-apache.m4 subversion-0.29.0-patch/build/ac-macros/svn-apache.m4
--- subversion-0.29.0/build/ac-macros/svn-apache.m4	2003-09-17 11:01:43.000000000 -0400
+++ subversion-0.29.0-patch/build/ac-macros/svn-apache.m4	2003-09-17 13:56:30.000000000 -0400
@@ -111,7 +111,7 @@
 
     APXS_CC="`$APXS -q CC`"
     APACHE_INCLUDES="$APACHE_INCLUDES -I$APXS_INCLUDE"
-    APACHE_LIBEXECDIR="`$APXS -q libexecdir`"
+    APACHE_LIBEXECDIR="`$APXS -q libexecdir`-extramodules"
 
     INSTALL_APACHE_RULE=install-mods-shared
 
