--- trunk/doc/book/tools/fo-stylesheet.xsl	2003-09-21 01:46:21.000000000 -0400
+++ trunk/doc/book/tools/fo-stylesheet.xsl	2003-09-21 01:46:21.000000000 -0400
@@ -1,6 +1,6 @@
 <xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" version='1.0'>
 
-  <xsl:import href="xsl/fo/docbook.xsl"/>
+  <xsl:import href="XSL_DIR/fo/docbook.xsl"/>
 
   <xsl:param name="fop.extensions" select="1" />
 
