Index: Makefile.in
===================================================================
--- subversion/Makefile.in	(revision 4874)
+++ subversion/Makefile.in	(working copy)
@@ -230,7 +230,6 @@
 
 ### HACK!! Find a better way to do this
 revision-install:
-	subversion/svnversion/svnversion $(top_srcdir) /repos/svn/trunk > $(DESTDIR)$(includedir)/subversion-1/svn-revision.txt
 
 install-static: @INSTALL_STATIC_RULES@
 
